### PR TITLE
Fix Paths (And Shapes) Stretching Modes + Add doc about shapes & brushes

### PR DESCRIPTION
**Pull request by carldebilly**
GitHub Issues: #387 #2 

## Bugfix
Path `Stretch` property wasn't working properly on WASM.

## What is the current behavior?
All stretch modes was rendering as if `Stretch.None` was used.

## What is the new behavior?
All 4 stretch modes now working properly as we can expect.

## PR Checklist

- [x] Tested code with current [supported SDKs](../README.md#supported)
- [x] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [x] Tests for the changes have been added (for bug fixes / features) (if applicable)
- [x] Contains **NO** breaking changes
- [x] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [X] Associated with an issue (GitHub or internal)

Internal Issue (If applicable):
* https://nventive.visualstudio.com/Umbrella/_workitems/edit/140035/

### CODE REVIEW

File: src/SamplesApp/SamplesApp.Shared/Samples/Paths/PathsPage.xaml [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+	xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+	xmlns:local="using:SamplesApp.Samples.Paths"
+	xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
+	xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+	mc:Ignorable="d"
+	Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">
+
+	<ScrollViewer>
+		<StackPanel>
+			<!--<TextBlock FontWeight="ExtraBold">None</TextBlock>
```
Comment (MatFillion) [inline src/SamplesApp/SamplesApp.Shared/Samples/Paths/PathsPage.xaml]: Any reason is the sample mostly commented out?

File: doc/articles/path-and-shapes.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+| `LinearGradientBrush`      | Yes     | Yes (2) | Yes (2) | [Documentation](https://docs.microsoft.com/en-us/uwp/api/Windows.UI.Xaml.Media.LinearGradientBrush) |
+| `SolidColorBrush`          | Yes     | Yes     | Yes     | [Documentation](https://docs.microsoft.com/en-us/uwp/api/Windows.UI.Xaml.Media.SolidColorBrush) |
+| `WebViewBrush`             | No      | No      | No      | [Documentation](https://docs.microsoft.com/en-us/uwp/api/Windows.UI.Xaml.Controls.WebViewBrush) |
+| `XamlCompositionBrushBase` | No      | No      | No      | [Documentation](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.media.xamlcompositionbrushbase) |
+
+Notes:
+
+1. `ImageBrush` on iOS can only be used as `Fill` brush, not supported for `Stroke` and the image needs to be a local asset.
+
+   `ImageBrush` on Android
```
Comment (carldebilly) [inline doc/articles/path-and-shapes.md]: I'll fix this

File: src/Uno.UI/UI/Xaml/Shapes/ArbitraryShapeBase.wasm.cs [hunk 3]
```diff
@@ -17,5 +19,147 @@ private IDisposable BuildDrawableLayer()
...
+			var measurements = GetMeasurements(finalSize);
+
+			foreach (FrameworkElement child in GetChildren())
+			{
+				var scaleX = measurements.scaleX.ToString(CultureInfo.InvariantCulture);
+				var scaleY = measurements.scaleY.ToString(CultureInfo.InvariantCulture);
+				var translateX = measurements.translateY.ToString(CultureInfo.InvariantCulture);
+				var translateY = measurements.translateX.ToString(CultureInfo.InvariantCulture);
+
+				child.SetStyleArranged(("transform", $"scale({scaleX}, {scaleY}) translate({translateX}px, {translateY}px)"));
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Shapes/ArbitraryShapeBase.wasm.cs]: We'll probably need to make an optimized method for that, parsing doubles is not particularly efficient.

File: src/Uno.UI/UI/Xaml/Shapes/ArbitraryShapeBase.wasm.cs [hunk 4]
```diff
@@ -17,5 +19,147 @@ private IDisposable BuildDrawableLayer()
...
+			}
+
+			return finalSize;
+		}
+
+		private (Size desiredSize, double translateX, double translateY, double scaleX, double scaleY) GetMeasurements(Size availableSize)
+		{
+			// Get the content size using Bounding Box on it.
+			var contentBBox = GetBBoxOfChildrenWithStrockThickness();
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Shapes/ArbitraryShapeBase.wasm.cs]: Stroke

Comment (carldebilly) [reply src/Uno.UI/UI/Xaml/Shapes/ArbitraryShapeBase.wasm.cs]: Oooh nice catch.

Comment (carldebilly) [reply src/Uno.UI/UI/Xaml/Shapes/ArbitraryShapeBase.wasm.cs]: Fixed.

File: src/Uno.UI/UI/Xaml/Shapes/ArbitraryShapeBase.wasm.cs (hunk 3 above)
Comment (carldebilly) [reply src/Uno.UI/UI/Xaml/Shapes/ArbitraryShapeBase.wasm.cs]: We need to set text for CSS anyway... so I don't see what you can optimize here.

**Pull request merged**